streetcrm/widgets.py

```diff
@@ -73,26 +73,36 @@ class TwelveHourTimeWidget(forms.MultiWidget):
         (12, "12"),
     )
 
+    MINUTE_CHOICES = (
+        ("", "--"), # Blank - no time specified.  This defaults to 00
+        (0, "00"),
+        (15, "15"),
+        (30, "30"),
+        (45, "45")
+    )
+
     TIME_SUFFIXES = (
         (AM_SUFFIX, "am"),
         (PM_SUFFIX, "pm"),
     )
 
     def __init__(self, attrs=None):
-        time = forms.Select(choices=self.TIME_CHOICES)
+        hour = forms.Select(choices=self.HOUR_CHOICES)
+        minute = forms.Select(choices=self.MINUTE_CHOICES)
         suffix = forms.RadioSelect(choices=self.TIME_SUFFIXES)
 
-        super(TwelveHourTimeWidget, self).__init__((time, suffix), attrs)
+        super(TwelveHourTimeWidget, self).__init__((hour, minute, suffix), attrs)
 
     def decompress(self, value):
         # Check for no value and return back
         if value is None:
-            return [None, None]
+            return [None, None, None]
 
         hours = int(value.strftime("%I"))
+        minutes = value.minute - (value.minute % 15)
```

Comment:
What's the modulo doing here?  I tried to figure it out and failed; a comment would help.

Reply:
Just rounding down to nearest 15 minutes, in case someone makes a post without using the interface.

Reply:
N % 15 doesn't round something to the nearest 15 minutes, though.  It just gives the remainder when you divide N by 15.  I.e., if you were to round 7:18pm to the "nearest 15 minutes", that'd be 7:15pm.  But modulo would be: 18 % 15 == 3, which would produce 7:03pm, which is not what you intended, I assume.

Reply:
Yes, but there's a subtraction there too.

Reply:
OH, my bad -- I see it now.  I misread the expression.  It's "VAL *minus* (VAL % 15)".  Okay, looks good, nothing to see here, move along folks, move along... :-).